### PR TITLE
Change TOC generation module to sigs.k8s.io/mdtoc

### LINKED ISSUE #1010: Evaluate mdtoc as replacement for our implementation
I recently found out that there is already a markdown TOC generator used in k/enhancements: https://github.com/kubernetes-sigs/mdtoc

We should check if this fits our needs and eventually replace the implementation we currently have.

Any volunteers? :innocent:  (If not I'll do it in the next weeks.)

### PR DESCRIPTION
**Pull request by puerco**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:

Until now, the kubernetes changelog and release-notes packages ran using the TOC generator written by @saschagrunert .
This PR removes it in favor of the newly-donated `sigs.k8s.io/mdtoc` to generate the table of contents in both tools.

#### Which issue(s) this PR fixes:

Fixes #1010

#### Special notes for your reviewer:

- Test data was updated with output from the new generator (has only minor changes in links and indentation).
- The old toc.go and tests have been removed.

Some samples generated with the new module are here:  https://github.com/puerco/lab/tree/master/release-notes/mdtoc

#### Does this PR introduce a user-facing change?

```release-note
- Switch to @tallclair 's TOC (`sigs.k8s.io/mdtoc`) to generate the table of contents in `krel changelog` and `release-notes`
```

### CODE REVIEW

File: cmd/krel/cmd/changelog.go [hunk 1]
```diff
@@ -313,7 +314,8 @@ func (c *Changelog) run(opts *changelogOptions, rootOpts *rootOptions) error {
 	}
 
 	logrus.Info("Generating TOC")
-	toc, err := notes.GenerateTOC(markdown)
+	toc, err := mdtoc.GenerateTOC([]byte(markdown))
+	fmt.Println(toc)
```
Comment (saschagrunert) [inline cmd/krel/cmd/changelog.go:318]: Can we remove this line?

File: go.mod [hunk 2]
```diff
@@ -18,18 +18,22 @@ require (
...
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.6.1
 	github.com/yuin/goldmark v1.2.1
-	golang.org/x/net v0.0.0-20200930145003-4acb6c075d10
+	golang.org/x/crypto v0.0.0-20201012173705-84dcc777aaee // indirect
+	golang.org/x/net v0.0.0-20201016165138-7b1cca2348c0
 	golang.org/x/oauth2 v0.0.0-20200902213428-5d25da1a8d43
-	google.golang.org/api v0.32.0
+	golang.org/x/sys v0.0.0-20201016160150-f659759dc4ca // indirect
+	golang.org/x/tools v0.0.0-20201016202518-de28cda03d9c // indirect
```
Comment (saschagrunert) [inline go.mod:33]: Let's remove those indirect dependencies and run `go mod tidy`

**Review by saschagrunert — APPROVED** [review]
Just two nits, otherwise LGTM :point_down:

File: cmd/krel/cmd/changelog.go (hunk 1 above)
Comment (puerco) [reply cmd/krel/cmd/changelog.go]: Yep, sorry. It was to debug and forgot to remove it. DOne.

File: go.mod (hunk 2 above)
Comment (puerco) [reply go.mod]: I've removed it. TBH I don't know what pulled those, but they are now gone.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**